sansio_lsp_client/structs.py

```diff
@@ -24,6 +24,10 @@ class Response(BaseModel):
     result: t.Optional[JSONDict]
     error: t.Optional[JSONDict]
 
+# type checked in Client._handle_response()
+class ResponseList(Response):
```

Comment:
Is the inheritance here just for type checks to pass, or do we really need it?

Reply:
I'm not quite sure what you mean

`ResponseList` is a `Response` but with a `list` `result`, inheritence seems entirely appropriate. 

pydantic was unable to parse such responses from this class:
``` python
class Response(BaseModel):
    id: t.Optional[Id]
    result: t.Optional[t.Union[
        t.List[t.Any],
        JSONDict,
    ]]
    error: t.Optional[JSONDict]
```

Does this answer your question?

Reply:
Please don't use inheritance. If `ResponseList` inherits from `Response`, then any code that works with a `Response` should also work with `ResponseList`. That is not true if `ResponseList` has a dict while `Response` has a list.

Maybe we could do something like [this](https://github.com/PurpleMyst/sansio-lsp-client/blob/96b585edfc8dcef562a80225404246a006834f34/sansio_lsp_client/client.py#L133-L144)?

Reply:
> pydantic was unable to parse such responses from this class:
> 
> ```python
> class Response(BaseModel):
>     id: t.Optional[Id]
>     result: t.Optional[t.Union[
>         t.List[t.Any],
>         JSONDict,
>     ]]
>     error: t.Optional[JSONDict]
> ```

I just tested this, and now it works (event on the lowest pydantic version, so it was probably my mistake somehow). Will this `Response` be ok? (and removing `ResponseList`)

Reply:
It's ok. We can change it later if it seems annoying.

Reply:
Speaking of testing, can you update `tests.py`? Currently I have no idea how you are testing your changes. Having them tested automatically would be a lot better.